packages/server/src/product.repo.ts

```diff
@@ -0,0 +1,35 @@
+import {Collection, Db, ObjectId, WithId} from "mongodb";
+import {Product} from "./types";
+
+type MongoProduct = Omit<Product, "id">;
+const docToProduct = ({_id, ...rest}: WithId<MongoProduct>) => ({id: _id.toString(), ...rest});
+
+export class MongoDBProductRepository {
+    private products: Collection<MongoProduct>;
+
+    constructor(db: Db) {
+        this.products = db.collection("products");
```

Comment:
why?

Reply:
1. This will prevent us from manually typing strings that only deal with logic
2. Typescript will prevent you from directly typing values and you'll need to use `COLLECTIONS.ORDERS etc..` directly (could be destructured
3. This could be imported to the test suit if needed 
4. Refactoring names is done in a single place.

Reply:
2 - how can TS prevent me from typing magic strings into `db.collection()`?
3 - don't need to - the DB tests are agnostic to the DB implementation
4 - can use a string const for that

Reply:
3. which types? the enum?
4. it's an implementation detail of how the mongo repo is built and is not important IMO. You may submit a PR for this if you wish :)